src/codecs/cats/whisker.c

```diff
@@ -0,0 +1,125 @@
+#include <string.h>
+#include <assert.h>
+
+#include "whisker.h"
+
+#ifdef RS41
+#define HARDWARE_ID 0x7d00
+#endif
+#ifdef DFM17
+#define HARDWARE_ID 0x7d01
+#endif
+
+#define SOFTWARE_ID 0x00
+
+static inline void cats_push_f16(cats_packet *p, float val)
+{
+    union {
+        float f;
+        uint32_t u;
+    } fu = { .f = val };
+
+    assert(sizeof fu.f == sizeof fu.u);
+
+    uint32_t f32 = fu.u;
+    uint32_t sign = (f32 >> 16) & 0x8000;
+    uint32_t exponent = ((f32 >> 23) - 127 + 15) & 0x1F;
+    uint32_t mantissa = f32 & 0x007FFFFF;
+    uint32_t f16 = sign | (exponent << 10) | (mantissa >> 13);
+
+    p->data[p->len++] = f16;
+    p->data[p->len++] = f16 >> 8;
+}
+
+static inline void cats_push_f32(cats_packet *p, float val)
+{
+    union {
+        float f;
+        uint32_t u;
+    } fu = { .f = val };
+
+    assert(sizeof fu.f == sizeof fu.u);
+
+    uint32_t f32 = fu.u;
+    p->data[p->len++] = f32;
+    p->data[p->len++] = f32 >> 8;
+    p->data[p->len++] = f32 >> 16;
+    p->data[p->len++] = f32 >> 24;
+}
+
+static inline void cats_push_u16(cats_packet *p, uint16_t val)
+{
+    for(int i = 0; i < 2; i++) {
+        p->data[p->len++] = val >> (i * 8);
+    }
+}
+
+static inline void cats_push_u32(cats_packet *p, uint32_t val)
+{
+    
+    for(int i = 0; i < 4; i++) {
+        p->data[p->len++] = val >> (i * 8);
+    }
+}
+
+void cats_append_identification_whisker(cats_packet *packet, char *callsign, uint8_t ssid, uint16_t icon)
+{
+    packet->data[packet->len++] = 0x00; // type = identidication
```

Comment:
Typo in `identidication`

Reply:
Also, there could be (#define) constants for the types (and other "magic" numbers)